testgen/generators/collation_short.py

```diff
@@ -92,47 +96,46 @@ def insert_collation_header(self, test_objs):
                 "UCA conformance test. Compare the first data string with the second and with strength = identical level (using S3.10). If the second string is greater than the first string, then stop with an error."
             )
 
-    def reset_test_data(self, rules, locale, attributes, strength):
-        rules = []
-        locale = ""
-        strength = None
-
-    def parse_compare(self, line_index, lines):
+    # ??? Pass in the attributes defined, adding them to each test ??
+    def check_parse_compare(self, line_index, lines):
         # Handles lines in a compare region
         # Test sections ("* compare") are terminated by
         # definitions of new collators, changing attributes, or new test sections.
+
         tests = []
         line_in = lines[line_index]
         if not self.compare_pattern.match(line_in):
-            return None, line_index
-
+            return None, line_index, None
+
+        # Patterns that end the processing of this "* compare" section
+        breakout_patterns = [self.compare_pattern,
+                             self.locale_string,
+                             self.root_locale,
+                             self.reorder_test,
+                             self.rule_header_pattern,
+                             self.test_line
+                             ]
+
+        # The tests constructed
+        tests = []
+        string2_errors = []  # line numbers and values that have conversion problems
         string1 = ''
         line_index += 1
         while line_index < len(lines):
             line_in = lines[line_index]
 
-            if self.compare_pattern.match(line_in):
-                break
-            if self.locale_string.match(line_in):
-                break
-            if self.root_locale.match(line_in):
-                break
-            if self.attribute_test.match(line_in):
-                break
-            if self.reorder_test.match(line_in):
-                break
-            if self.rule_header_pattern.match(line_in):
-                break
-            if self.test_line.match(line_in):
+            # Time to end this set of comparison tests.
+            if any([p.match(line_in) for p in breakout_patterns]):
                 break
 
             # It's a blank line, comment, or a comparison
-            # ignore comment at end of this rulrule_match.group(1)e line
+            # ignore comment at end of this comment
             if line_in == '' or line_in[0] == '#':
-                # Skip blank lines
+                # Skip blank and comment lines
                 line_index += 1
                 continue
 
+            # A comparison line starts with either "<" or "="
```

Comment:
This comment belongs next to the regex because it explains why the regex is written the way it is. Copy-paste there. And optionally, reword this line to be more generic (ex: "check for a comparison"). That way, if the file format changes, and the regex pattern changes, you don't need to change this line b/c there's no duplication of content that leads to inconsistency & staleness.